### PR TITLE
update project folder structure

### PR DESCRIPTION
**Pull request by shamshaw**
#3 
Updating project organization and folder structure

### CODE REVIEW

**Comment by shamshaw** [comment]
Script has been decomposed into separate functions.  The example use of the functions is shown below:
```
library(dplyr)
library(readr)
library(stringr)
library(sbtools)
library(whisker)

source("1_fetch/src/fetch_data.R")
model_data <- fetch_data()

source("2_process/src/prep_data.R")
eval_data <- prep_data(model_data)
make_log_file(eval_data)

source("3_visualize/src/viz_data.R")
plot_model_compare(eval_data)
```

**Comment by shamshaw** [comment]
Additional note: original code structure of functions has been preserved at this point assuming additional code revisions will be integrated into future steps

File: 2_process/src/make_log_file.R [hunk 1]
```diff
@@ -0,0 +1,19 @@
+make_log_file <-function(eval_data)
```
Comment (lindsayplatt) [inline 2_process/src/make_log_file.R]: I would suggest wrapping the function contents with `{}`. Works without it but better to be explicit!

Comment (lindsayplatt) [reply 2_process/src/make_log_file.R]: Although, now that I am looking at this and your other files. I think that you may have meant to use the `make_log_file()` function that you currently have in `2_process/src/prep_data.R` here instead (it does not rely on the object `project_output_dir` which does not exist + uses the `{}`).

**Review by lindsayplatt — COMMENTED** [review]
I think this will come with the other steps in this module, but it is usually best practices to pass in filepaths as arguments to the function so that users can see what file is being saved without digging into the function guts. For this reason, I would suggest setting them up like this:

```r
# Defining the function
fetch_data <- function(out_file) {
  ...
}

# Using the function
fetch_data('model_RMSEs.csv')
```

File: 2_process/src/make_log_file.R (hunk 1 above)
Comment (shamshaw) [reply 2_process/src/make_log_file.R]: Oops, yes I meant delete that make_log_file.R but looks like I pushed it up. I'll remove that. In matlab it is much more common that each function is its own file, but I think in R it is fairly common to group related functions in one file?

Comment (shamshaw) [reply 2_process/src/make_log_file.R]: Updated this PR with the make_lot_file.R file removed

Comment (lindsayplatt) [reply 2_process/src/make_log_file.R]: I see it both ways and I don't have a good answer for when you should do one file per function vs group related functions into files. My own personal preference is that longer functions get their own files, but smaller functions can exist with others that do similar things or operate on similar data.

**Review by lindsayplatt — APPROVED** [review]

**Comment by shamshaw** [comment]
> I think this will come with the other steps in this module, but it is usually best practices to pass in filepaths as arguments to the function so that users can see what file is being saved without digging into the function guts. For this reason, I would suggest setting them up like this:
> 
> ```r
> # Defining the function
> fetch_data <- function(out_file) {
>   ...
> }
> 
> # Using the function
> fetch_data('model_RMSEs.csv')
> ```

Got it - this makes sense to me and what I often do - I'll incorporate this change into later steps

**Pull request merged**